### ISSUE TITLE
Are we able to use string instead of new Date() ?

### ISSUE THREAD

**Issue by tombohub** [RobinBlomberg/kysely-codegen#123]
postgres generated code for table is:

```ts
export interface OvernightDailyOvernightPerformance {
  ...
  date: Timestamp;
  ...
}
```

I want to filter by date:

```ts
const rows = await db
    .selectFrom("overnight.daily_overnight_performance")
    .where("date", "=", "2023-06-02")
    .execute()
```

but it's an error: 
![image](https://github.com/RobinBlomberg/kysely-codegen/assets/3029412/0febfb30-5f89-44f1-bd88-2efa08ba61b6)

and cannot run the script.

If I change Timestamp type from:

```ts
export type Timestamp = ColumnType<Date, Date | string, Date | string>;
```

to:

```ts
export type Timestamp = ColumnType<Date | string, Date | string, Date | string>;
```

it still shows the error: 
![image](https://github.com/RobinBlomberg/kysely-codegen/assets/3029412/d80a0035-3d3c-4074-99aa-ce9cc2a8f7c5)

but at least I can run the script.

Can you make it so it generates code where date string is acceptable for selection?

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/RobinBlomberg) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/RobinBlomberg/kysely-codegen/issues/123">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/123/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/123/pledge.svg">
</picture>
</a>

**Comment by SangJunBak** [comment]
I too have this problem!

**Comment by RobinBlomberg** [comment]
This sounds like a duplicate of #121. Someone posted a solution there:

```ts
import { Selectable } from 'kysely'

const results: Selectable<Thing> = await db.selectFrom('thing').selectAll().execute()
```

Please check if that solves the issue. If so, I will add the solution to the README.

**Comment by koskimas** [comment]
@RobinBlomberg This is a different issue. The issue is that `where` uses the `Selectable` type, which is `Date` in `Timestamp`.

@tombohub
> it still shows the error:

No it doesn't. Restart vscode or reboot its typescript server. If you're able to compile that typescript, then surely the typescript running inside vscode also accepts the code. You just didn't update something correctly.

There's probably nothing kysely-codegen should do here. If you change the selectable type, then the result type is also `Date | string`. What you (the user) can do is create a helper function like this:

```ts
function toDate(date: string) {
  return sql.val<Date>(date)
}

const rows = await db
    .selectFrom("overnight.daily_overnight_performance")
    .where("date", "=", toDate("2023-06-02"))
    .execute()
```

**Comment by RobinBlomberg** [comment]
Should be fixed now in [kysely-codegen@0.17.0](https://www.npmjs.com/package/kysely-codegen)! Use the new `--date-parser` option.
